oracle/config/transfer-watcher.config.js

```diff
@@ -29,6 +29,11 @@ if (!transferWatcherRequired) {
   process.exit(EXIT_CODES.WATCHER_NOT_REQUIRED)
 }
 
+const workerQueueConfig = {}
+if (baseConfig.id === 'erc-native') {
+  workerQueueConfig.workerQueue = 'convert-to-chai'
```

Comment:
I cannot find where it is being used... Could you help me?

Reply:
It is used in `transfer-watcher` for `ERC_TO_NATIVE` mode.
When the corresponding transfer event is caught, a message is sent to `convert-to-chai` queue, so that a validator will check if there is a need to call `convertDaiToChai`.

Reply:
Ok. I see just now. Quite unexpected for me that the watcher did not reported with an error about absence of `config.workerQueue` previously.

Reply:
Yes, this config usage is a little hard to follow. It was built having in mind that not all watchers have a side worker to which push jobs beside of the sender. The first one was `half-duplex-watcher` to push jobs to the `swap-tokens` worker. So only if `workerQueue` is defined in the config, it will try to push a job to that queue, that's why there is no error present.

https://github.com/poanetwork/tokenbridge/blob/bc6dd131939f5995d3e1ee95843026ad9ddf5134/oracle/src/services/amqpClient.js#L26-L28

https://github.com/poanetwork/tokenbridge/blob/bc6dd131939f5995d3e1ee95843026ad9ddf5134/oracle/src/watcher.js#L189-L191

I hope this helps to clarify how it works

Reply:
Thanks

Reply:
Consider to revisit logic in `transfer-watcher` to avoid sending of messages to the worker queue if the contracts are still not updated (has no support of Chai token at all) or earning interest with Chai is deactivated.